helper/hex/hex_test.go

```diff
@@ -0,0 +1,38 @@
+package hex
+
+import (
+	"crypto/rand"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+// TestEncodeDecodeUint64 verifies that uint64 values
+// are properly encoded and decoded from hex
+func TestEncodeDecodeUint64(t *testing.T) {
+	t.Parallel()
+
+	generateRandomNumbers := func(count int) []uint64 {
+		numbers := make([]uint64, count)
+
+		for index := 0; index < count; index++ {
+			randNum, _ := rand.Int(
+				rand.Reader,
+				big.NewInt(int64(count)),
+			)
+
+			numbers[index] = randNum.Uint64()
+		}
+
+		return numbers
+	}
+
+	for _, value := range generateRandomNumbers(100) {
+		encodedValue := EncodeUint64(value)
+
+		decodedValue, err := DecodeUint64(encodedValue)
+		assert.NoError(t, err)
+
+		assert.Equal(t, value, decodedValue)
+	}
```

Comment:
We never hardcoded number values before in tests, nor should we - these numbers truly need to be random as we're not testing random number generation but encoding and decoding.

The functionality that was added as part of this PR (header encoding / decoding) is covered in another test

Reply:
I don't know why we need to test with random value...

There are 2 things I want to say

1. What is the purpose of the test? Do we need to test whether go can calculate `1+1` correctly? Or course, we don't need this. Likewise, it's not important to test `strings.TrimPrefix` and `strconv.ParseUint` work correctly. The important thing is to check the DecodeUint64 can parse all of the values in the range we expect and return the correct value. (Otherwise, this function returns error or not) In such a case, do we need to test with random value? In such a case, we should do boundary value test. (In addition, we should give the value with wrong format to check the function returns error) You said "We never hardcoded number values" but the basis of unit test is to give sample inputs and check whether the outputs equal to expected values.

2. Why does new test encode and decode? The new code is to decode only. We should add the test to check the function first. But this test does both encode and decode. We should thing about unit test, e2e test, and fuzzy test separately. The test that checks many things is hard to read and difficult to fix.
 
And I'm not talking about the test for header encoding / decoding. But this test encodes uint64 values and decodes it.

Reply:
@zivkovicmilos

Reply:
Got it, no problem in keeping it minimal, you're right that we double verify stuff here.

Fixed:
[fe561e7](https://github.com/0xPolygon/polygon-edge/pull/642/commits/fe561e7fee2ca2975e121541ea2d65ce5a1ddee3)

Reply:
Thank you for fixing!